geode-core/src/upgradeTest/java/org/apache/geode/security/AuthExpirationDUnitTest.java

```diff
@@ -109,4 +116,92 @@ public void clientShouldReAuthenticateWhenCredentialExpiredAndOperationSucceed()
     assertThat(region.size()).isEqualTo(2);
   }
 
+  @Test
+  public void userShouldReAuthenticateWhenCredentialExpiredAndOperationSucceed() throws Exception {
+    int serverPort = server.getPort();
+    ClientVM clientVM = lsRule.startClientVM(0, clientVersion,
+        c -> c.withMultiUser(true)
+            .withProperty(SECURITY_CLIENT_AUTH_INIT, UpdatableUserAuthInitialize.class.getName())
+            .withPoolSubscription(true)
+            .withServerConnection(serverPort));
+
+    clientVM.invoke(() -> {
+      UpdatableUserAuthInitialize.setUser("user1");
+      ClientCache clientCache = ClusterStartupRule.getClientCache();
+      assert clientCache != null;
+      clientCache.createClientRegionFactory(ClientRegionShortcut.PROXY).create("region");
+      Properties userSecurityProperties = new Properties();
+      userSecurityProperties.put("security-username", "serviceUser0");
+      userSecurityProperties.put("security-password", "serviceUser0");
+      RegionService regionService = clientCache.createAuthenticatedView(userSecurityProperties);
+      Region<Object, Object> region = regionService.getRegion("/region");
+      region.put(0, "value0");
+    });
+
+    clientVM.invoke(() -> {
+      ClientCache clientCache = ClusterStartupRule.getClientCache();
+      Properties userSecurityProperties = new Properties();
+      userSecurityProperties.put("security-username", "serviceUser1");
+      userSecurityProperties.put("security-password", "serviceUser1");
+      assert clientCache != null;
+      RegionService regionService = clientCache.createAuthenticatedView(userSecurityProperties);
+      Region<Object, Object> region = regionService.getRegion("/region");
+      region.put(1, "value1");
+      regionService.close();
```

Comment:
I would think user would continue to use same regionService and expect the next put to succeed. (the whole expire user and re-authenticate happens behind the scene and the client is doing operations like nothing happened).

Reply:
I agree. I'll see if I can find reuse the same `regionService` from one `clientVM` invocation to another. If you have ideas on how to accomplish this I'm open to suggestions.

Reply:
I was unable to reuse the `regionService` but came up with a way accomplish the automatic re-authentication behind the scenes with changes to the `AuthInitialize` implementation we're using for the tests.